x/concentrated-liquidity/swaps.go

```diff
@@ -210,6 +210,7 @@ func (k Keeper) calcOutAmtGivenIn(ctx sdk.Context,
 	priceLimit sdk.Dec,
 	poolId uint64,
 ) (tokenIn, tokenOut sdk.Coin, updatedTick sdk.Int, updatedLiquidity, updatedSqrtPrice sdk.Dec, err error) {
+	ctx, _ = ctx.CacheContext()
```

Comment:
We should return the second return of `CacheContext` from the caller (`calcOutAmtGivenIn`) itself. The second return is a callback to write the cache context.

The reason for this requirement is that if `calcOutAmtGivenIn` is called from `swapOtAmtGivenIn`, we actually do want to write the cache context to storage.

Reply:
I also noticed that part in the original issue, but did not quite understand. 

What's the reason we want to use write cache context here? Do we not store in state anyways using the results from calling the calc method anyways within the swap method?(https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/swaps.go#L167)

Reply:
I am not sure if this is what you are saying Roman, but shouldnt we pass this cache ctx as a return value and then write in the swap function that calls this? If that is what you are saying, I agree. Without doing this, despite us calling applySwap, the tick boundaries that got passed wont be updated.

Reply:
Gotcha, this makes sense